### DISCUSSION TITLE
[BUG] longhorn default disk name duplicating across nodes

### DISCUSSION THREAD

**Discussion by migs35323** [longhorn/longhorn#9592] [General]
## Describe the bug

I cloned my basic machine in my virtual enviroment (a machine that has ubuntu and a few dependencies installed before) to create a new cluster node in another physical disk
this cloned vm has diferent hostname however some details like the virtual HDD id and machine id are very likely the same...
the issue is, it is generating a default name for the disk that is the same as the first machine, i did deleted the `/var/lib/longhorn` folder deleted the node and registered it as a new one, however the generated name is the same...

this created some issues, the longhorn UI is cycling the replicas state between the 2 machines and being confused about affinity

IS there a simple explanation how this longhorn disk name is being generated? i can't find the corresponding "uuid" anywhere, like a part of the linux disk UUID or label or something

## Environment

 - Longhorn version: 1.7.1
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: k3s 1.28
   - Number of control plane nodes in the cluster: 3
   - Number of worker nodes in the cluster: 5-6
 - Node config
   - OS type and version: ubuntu 22.04 cloud img

   - Disk type (e.g. SSD/NVMe/HDD): one of the affected nodes is a nvme, other is a 2.5' ssd
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): vms under proxmox

**Comment by c3y1huang** [comment]
The default disk name is generated with [DefaultDiskPrefix + diskStat.DiskID](https://github.com/longhorn/longhorn-manager/blob/v1.7.1/types/types.go#L1087), where `diskStat.DiskID` is coming from [GetDiskStat(dataPath)](https://github.com/longhorn/go-common-libs/blob/a0195307257b52d182e4c07bf9db76f5aef25f43/io/file.go#L226). To obtain the same disk ID, you can use the following command:
```bash
stat -fc "{\"path\":\"%n\",\"fsid\":\"%i\",\"type\":\"%T\",\"freeBlock\":%f,\"totalBlock\":%b,\"blockSize\":%S}" /var/lib/longhorn/
```
The `fsid` value should be what you are looking for.
